mne/encoding/feature.py

```diff
@@ -0,0 +1,319 @@
+        matrix. Defaults to False.
+    """
+    def __init__(self, n_times, sfreq=1., sparse=False):
+        self.n_times = n_times
+        self.sfreq = sfreq
+        self.feature_type = 'event_id'
+        self.sparse = sparse
+
+    def fit(self, event_ixs, event_ids=None, event_dict=None):
+        """Binarize the events.
+
+        Parameters
+        ----------
+        event_ixs : array, shape (n_events)
+            The event indices or onsets in seconds. `If self.sfreq == 1`,
+            then these are assumed to be indices. If `self.sfreq != 1`, these
+            should be onsets in seconds and will be multiplied by `sfreq`.
+        event_ids : array-like, shape (n_events) | None
+            The event ID for each event. If None, all events are assumed to be
+            of same type.
+        event_dict : dict
+            A dictionary of (event_id: event_id_name) pairs. Defines a string
+            name for each event type.
+        """
+        if event_ixs.ndim > 1:
+            raise ValueError("events must be shape (n_trials, 3),"
+                             " found shape %s" % str(event_ixs.shape))
+        event_ids = np.ones_like(event_ixs) if event_ids is None else event_ids
+        unique_event_types = np.unique(event_ids)
+
+        if event_dict is None:
+            event_dict = dict(('event_%s' % ii, ii)
+                              for ii in unique_event_types)
+        ev_dict_rev = dict((value, key) for key, value in event_dict.items())
+
+        # Turn event_ixs from seconds to indices in case sfreq != 1
+        event_ixs = (event_ixs * self.sfreq).astype(int)
+
+        # Iterate through event types and create columns of event onsets.
+        events_continuous = np.zeros([self.n_times, len(unique_event_types)])
+        event_names = []
+        for ii, ev_type in enumerate(unique_event_types):
+            msk_events = event_ids == ev_type
+            i_ev = event_ixs[msk_events]
+            events_continuous[i_ev, ii] = 1
+
+            # Handle event names
+            event_names.append(ev_dict_rev[ev_type])
+        if self.sparse is True:
+            events_continuous = sparse.csr_matrix(events_continuous)
+
+        self.names_ = event_names
+        self.events_continuous_ = events_continuous
+        self.unique_event_types_ = unique_event_types
+
+    def transform(self, X=None, y=None):
+        """Return continuous events data.
+
+        Returns
+        -------
+        events_continuous : array, shape (n_times, n_unique_events)
+            A binary array with one column for each unique event type. 1s
+            in the array correspond to event onsets for that row.
+        """
+        return self.events_continuous_
+
+    def fit_transform(self, event_ixs, event_ids=None, event_dict=None):
+        self.fit(event_ixs, event_ids=event_ids, event_dict=event_dict)
+        return self.transform()
+
+
+class DataSubsetter(object):
+    """Use for returning a subset of time.
+
```

Comment:
I was using it to subset data for cross-validation when you're extracting features as a part of the preprocessing pipeline. E.g., if you're using a DataDelayer object in the pipeline, then you want to do the delaying first, and then take the subset of data that you want to fit. I could see it also being used for decimation too, so model fitting goes faster, though there would be more ways for the person to mess up the analysis in that case...